src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/PersonMatchScoreRestClient.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.justice.probation.courtcasematcher.restclient;
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+import reactor.util.retry.Retry.RetrySignal;
+import uk.gov.justice.probation.courtcasematcher.restclient.model.personmatchscore.PersonMatchScoreRequest;
+import uk.gov.justice.probation.courtcasematcher.restclient.model.personmatchscore.PersonMatchScoreResponse;
+
+import java.time.Duration;
+import java.util.function.Predicate;
+
+@Component
+@Slf4j
+public class PersonMatchScoreRestClient {
+
+    @Setter
+    @Value("${person-match-score.post-match-url}")
+    private String postMatchUrl;
+
+    @Setter
+    @Value("${offender-search.max-retries:3}")
+    private int maxRetries;
+
+    @Setter
+    @Value("${offender-search.min-backoff-seconds:5}")
+    private int minBackOffSeconds;
+
+    private final WebClient webClient;
+
+    @Autowired
+    public PersonMatchScoreRestClient(@Qualifier("personMatchScoreWebClient") WebClient webClient) {
+        super();
+        this.webClient = webClient;
+    }
+    public Mono<PersonMatchScoreResponse> match(PersonMatchScoreRequest body){
+
+        return webClient.post()
+            .uri(postMatchUrl)
+                .body(BodyInserters.fromPublisher(Mono.just(body), PersonMatchScoreRequest.class))
+                .accept(MediaType.APPLICATION_JSON)
+                .retrieve()
+                .bodyToMono(PersonMatchScoreResponse.class)
+                .retryWhen(Retry.backoff(maxRetries, Duration.ofSeconds(minBackOffSeconds))
```

Comment:
I don't think we need to retry for this, at least not until we've established how well it performs. Can we configure it to 0 retries initially?

Reply:
updated to 0 ta